shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -249,11 +255,15 @@ - (void)popSystemNavigator:(BOOL)isAnimated {
   if (navigationController) {
     [navigationController popViewControllerAnimated:isAnimated];
   } else {
+#if !APPLICATION_EXTENSION_API_ONLY
     UIViewController* rootViewController =
         [UIApplication sharedApplication].keyWindow.rootViewController;
     if (engineViewController != rootViewController) {
```

Comment:
Not sure if we still need to check _something_ here in the share extension to avoid regressing #30939.